cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/ProgramScheduleStoreDataset.java

```diff
@@ -579,4 +607,9 @@ private static ScheduleId extractScheduleIdFromTriggerKey(String triggerRowKey)
     return Bytes.toBytes(
       Joiner.on(ROW_KEY_SEPARATOR).join(appId.getNamespace(), appId.getApplication(), appId.getVersion(), ""));
   }
+
+  private static byte[] keyPrefixForNamespaceScan(NamespaceId namespaceId) {
+    return Bytes.toBytes(
+      Joiner.on(ROW_KEY_SEPARATOR).join(namespaceId.getNamespace(), ""));
```

Comment:
easier to just use `String.format`

Reply:
this is following the pattern of constructing key prefix as in other methods

Reply:
well if there is a list, makes sense to use joiner, this is just one item.